src/posix/platform/rcp_caps_diag.cpp

```diff
@@ -498,22 +502,149 @@ void RcpCapsDiag::TestSpinelCommands(Category aCategory)
     }
 }
 
+void RcpCapsDiag::ProcessCapabilityFlags(void)
+{
+    TestRadioCapbilityFlags();
+    TestSpinelCapbilityFlags();
+}
+
+void RcpCapsDiag::TestRadioCapbilityFlags(void)
+{
+    otError      error;
+    unsigned int radioCaps;
+    SuccessOrExit(error = mRadioSpinel.Get(SPINEL_PROP_RADIO_CAPS, SPINEL_DATATYPE_UINT_PACKED_S, &radioCaps));
+
+    Output("\r\nRadio Capbility Flags :\r\n");
+
+    TestRadioCapbilityFlags(kCategoryThread1_1, static_cast<uint32_t>(radioCaps));
+    TestRadioCapbilityFlags(kCategoryThread1_2, static_cast<uint32_t>(radioCaps));
+    TestRadioCapbilityFlags(kCategoryUtils, static_cast<uint32_t>(radioCaps));
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        Output("Failed to get radio capability flags: %s", otThreadErrorToString(error));
+    }
+
+    return;
+}
+
+void RcpCapsDiag::TestRadioCapbilityFlags(Category aCategory, uint32_t aRadioCapFlags)
+{
+    static constexpr FlagEntry kRadioFlagEntries[] = {
+        {kCategoryThread1_1, OT_RADIO_CAPS_ACK_TIMEOUT},     {kCategoryThread1_1, OT_RADIO_CAPS_TRANSMIT_RETRIES},
+        {kCategoryThread1_1, OT_RADIO_CAPS_CSMA_BACKOFF},    {kCategoryThread1_2, OT_RADIO_CAPS_TRANSMIT_SEC},
+        {kCategoryThread1_2, OT_RADIO_CAPS_TRANSMIT_TIMING}, {kCategoryUtils, OT_RADIO_CAPS_ENERGY_SCAN},
+        {kCategoryUtils, OT_RADIO_CAPS_SLEEP_TO_TX},         {kCategoryUtils, OT_RADIO_CAPS_RECEIVE_TIMING},
+        {kCategoryUtils, OT_RADIO_CAPS_RX_ON_WHEN_IDLE}};
+
+    Output("\r\n%s :\r\n", CategoryToString(aCategory));
+
+    for (const FlagEntry &entry : kRadioFlagEntries)
+    {
+        if (entry.mCategory != aCategory)
+        {
+            continue;
+        }
+
+        OutputFormat(RadioCapbilityToString(entry.mFlag), SupportToString((aRadioCapFlags & entry.mFlag) > 0));
```

Comment:
```suggestion
        if (entry.mCategory == aCategory)
        {
            OutputFormat(RadioCapbilityToString(entry.mFlag), SupportToString((aRadioCapFlags & entry.mFlag) > 0));
        }
```

Avoid using continue as possible?

Reply:
Updated